vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridDataViewImpl.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2000-2020 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.vaadin.flow.component.grid.dataview;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import com.vaadin.flow.component.grid.Grid;
+import com.vaadin.flow.data.provider.AbstractDataView;
+import com.vaadin.flow.data.provider.DataCommunicator;
+import com.vaadin.flow.data.provider.DataProvider;
+import com.vaadin.flow.data.provider.IdentifierProvider;
+
+/**
+ * Implementation of generic data view for grid.
+ * 
+ * @param <T> the item type
+ * @since
+ */
+public class GridDataViewImpl<T> extends AbstractDataView<T>
+        implements GridDataView<T> {
+
+    private DataCommunicator<T> dataCommunicator;
+
+    public GridDataViewImpl(DataCommunicator<T> dataCommunicator,
+            Grid<T> grid) {
+        super(dataCommunicator::getDataProvider, grid);
+        this.dataCommunicator = dataCommunicator;
+    }
+
+    @Override
+    public Stream<T> getItems() {
+        return dataCommunicator.getDataProvider()
+                .fetch(dataCommunicator.buildQuery(0, Integer.MAX_VALUE));
+    }
+
+    @Override
+    public int getSize() {
+        return dataCommunicator.getDataSize();
+    }
+
+    @Override
+    public boolean contains(T item) {
+        final IdentifierProvider<T> identifierProvider = getIdentifierProvider();
+
+        Object itemIdentifier = identifierProvider.apply(item);
+        Objects.requireNonNull(itemIdentifier,
+                "Identity provider should not return null");
+        return getItems().anyMatch(
+                i -> itemIdentifier.equals(identifierProvider.apply(i)));
```

Comment:
I intentionally moved `identifierProvider.apply(i)` to the new line, to get it simplier to analyse the root cause if `identifierProvider.apply()` would throw an exception. Does it make sense?

Reply:
Then you should remember to use the //@formatter:off and //@formatter:on

Reply:
You're right. Thanks!